cbindings/pedersen.t.cc

```diff
@@ -120,13 +122,19 @@ compute_expected_ristretto255_commitment(const std::vector<T>& data,
 template <class T>
 static cg1t::compressed_element
 compute_expected_bls12_381_g1_commitment(const std::vector<T>& data,
-                                         const std::vector<cg1t::element_p2>& generators) {
+                                         const std::vector<cg1t::element_affine>& generators) {
   SXT_DEBUG_ASSERT(data.size() == generators.size());
 
+  // Convert from affine to projective elements
+  memmg::managed_array<cg1t::element_p2> generators_span_array(generators.size());
+  basct::span<cg1t::element_p2> generators_span{generators_span_array.data(), generators.size()};
+  basct::cspan<cg1t::element_affine> generators_span_affine{generators.data(), generators.size()};
```

Comment:
Why are there all these extra variables? Why do you need generators_span and generators_span_affine?

Reply:
I think it stems from a misuse of span on my part. I used references to the spans in the `batch_to_element_p2` function signature
```
void batch_to_element_p2(basct::span<cg1t::element_p2>& p, basct::cspan<cg1t::element_affine>& a)
```
and got the following compilation errors when I didn't create the extra variables `generators_span` and `generatros_span_affine`:
```
error: non-const lvalue reference to type 'span<...>' cannot bind to a temporary of type 'span<...>'
error: non-const lvalue reference to type 'cspan<...>' cannot bind to a temporary of type 'cspan<...>'
```

I have removed the references in the function signature and cleaned up the extra variables. Thanks.